### PR TITLE
introduce github action for ci test asan

### PR DESCRIPTION
**Pull request by LHT129**
- test asan on github action
- format check on github action
- other test on circleci

### CODE REVIEW

File: scripts/collect_cpp_coverage.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ROOT_DIR="${SCRIPTS_DIR}/../../"
+
+COVERAGE_DIR="${ROOT_DIR}/testresult/coverage"
+if [ -d "${COVERAGE_DIR}" ]; then
+    rm -rf "${COVERAGE_DIR:?}"/*
+else
+    mkdir -p "${COVERAGE_DIR}"
+fi
+
+# 通过lcov生成coverage info文件
```
Comment (wxyucs) [inline scripts/collect_cpp_coverage.sh]: comment in english

File: scripts/collect_cpp_coverage.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+     --exclude "*/vsag/include/vsag/expected.hpp*" \
+     --exclude "*_test.cpp" \
+     --capture \
+     --ignore-errors mismatch,mismatch \
+     --ignore-errors unused,unused \
+     --ignore-errors inconsistent,inconsistent \
+     --directory . \
+     --output-file  "${COVERAGE_DIR}/coverage_ut.info"
+
+# 合并覆盖率文件生成最终文件coverage.info
```
Comment (wxyucs) [inline scripts/collect_cpp_coverage.sh]: comment in english

Comment (LHT129) [reply scripts/collect_cpp_coverage.sh]: remove the comment

File: scripts/collect_cpp_coverage.sh (hunk 1 above)
Comment (LHT129) [reply scripts/collect_cpp_coverage.sh]: remove the comment

**Review by wxyucs — APPROVED** [review]
lgtm

File: src/quantization/sq8_uniform_quantizer.h [hunk 3]
```diff
@@ -89,20 +89,28 @@ SQ8UniformQuantizer<metric>::SQ8UniformQuantizer(int dim, Allocator* allocator)
     lower_bound_ = std::numeric_limits<DataType>::max();
     diff_ = std::numeric_limits<DataType>::lowest();
 
+    size_t align_size = 1;
+    if constexpr (metric == MetricType::METRIC_TYPE_L2SQR) {
+        align_size = std::max(align_size, sizeof(norm_type));
+    }
+    if constexpr (metric == MetricType::METRIC_TYPE_IP or
```
Comment (jiaweizone) [inline src/quantization/sq8_uniform_quantizer.h:96]: Irrelevant code?

File: .circleci/config.yml [hunk 4]
```diff
@@ -23,7 +23,6 @@ jobs:
     resource_class: medium+
     steps:
       - checkout
-      - run: ./scripts/check_format.sh
```
Comment (jiaweizone) [inline .circleci/config.yml:26]: why remove format check ?

Comment (LHT129) [reply .circleci/config.yml:26]: enable it on github action

File: src/quantization/sq8_uniform_quantizer.h (hunk 3 above)
Comment (LHT129) [reply src/quantization/sq8_uniform_quantizer.h:96]: fix asan problem, the github action use asan test

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by ShawnShawnYou — APPROVED** [review]

**Review by inabao — APPROVED** [review]
LGTM

**Pull request merged**